listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -130,6 +130,47 @@ export default class ListenCard extends React.Component<
     }
   };
 
+  recommendListenToFollowers = async () => {
+    const { listen, currentUser, isCurrentUser, newAlert } = this.props;
+
+    if (isCurrentUser && currentUser?.auth_token) {
+      const metadata: UserTrackRecommendationMetadata = {
+        artist_name: _get(listen, "track_metadata.artist_name"),
+        track_name: _get(listen, "track_metadata.track_name"),
+        release_name: _get(listen, "track_metadata.release_name"),
+        recording_mbid: _get(
+          listen,
+          "track_metadata.additional_info.recording_mbid"
+        ),
+        recording_msid: _get(
+          listen,
+          "track_metadata.additional_info.recording_msid"
+        ),
+        artist_msid: _get(listen, "track_metadata.additional_info.artist_msid"),
+      };
+
+      try {
+        const status = await this.APIService.recommendTrackToFollowers(
+          currentUser.name,
+          currentUser.auth_token,
+          metadata
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            `You recommended a track to your followers!`,
+            `${metadata.artist_name} - ${metadata.track_name}`
+          );
+        }
+      } catch (error) {
```

Comment:
This won't catch successful requests with an error code.
Currently if I receive a 500 error (error in the action, but successful request) nothing happens, no error is thrown by checkStatus in APIService.ts
An issue with checkStatus I think more than with your code, but until that's solved better to throw a helpful error yourself after `if (status === 200) {…}`.

Reply:
Scratch that ! The issue is in `APIService.recommendTrackToFollowers` where `this.checkStatus(response);` should be `await this.checkStatus(response);`